tools/cldr-code/src/main/java/org/unicode/cldr/test/CheckPlaceHolders.java

```diff
@@ -18,14 +34,150 @@
+            int firstSurname = Integer.MAX_VALUE;
+            int firstGiven = Integer.MAX_VALUE;
+
+            Multimap<Field, Integer> fieldToPositions = namePattern.getFieldPositions();
+            for (Entry<Field, Collection<Integer>> entry : fieldToPositions.asMap().entrySet()) {
+
+                // If a field occurs twice, probably an error. Could relax this upon feedback
+
+                Collection<Integer> positions = entry.getValue();
+                if (positions.size() > 1) {
+
+                    // However, do allow prefix&core together
+
+                    boolean skip = false;
+                    if (entry.getKey() == Field.surname) {
+                        Iterator<Integer> it = positions.iterator();
+                        Set<Modifier> m1 = namePattern.getModifiedField(it.next()).getModifiers();
+                        Set<Modifier> m2 = namePattern.getModifiedField(it.next()).getModifiers();
+                        skip = m1.contains(Modifier.core) && m2.contains(Modifier.prefix)
+                            || m1.contains(Modifier.prefix) && m2.contains(Modifier.core);
+                    }
+
+                    if (!skip) {
+                        result.add(new CheckStatus().setCause(this)
+                            .setMainType(CheckStatus.errorType)
+                            .setSubtype(Subtype.invalidPlaceHolder)
+                            .setMessage("Duplicate fields: " + entry));
+                    }
+                }
+
+                // gather some info for later
+
+                Integer leastPosition = positions.iterator().next();
+                switch (entry.getKey()) {
+                case given: case given2:
+                    firstGiven = Math.min(leastPosition, firstGiven);
+                    break;
+                case surname: case surname2:
+                    firstSurname = Math.min(leastPosition, firstSurname);
+                    break;
+                default: // ignore
+                }
+            }
+
+            // we must have at least a given* and a surname*
+
+            if (firstGiven == Integer.MAX_VALUE && firstSurname == Integer.MAX_VALUE) {
+                result.add(new CheckStatus().setCause(this)
+                    .setMainType(CheckStatus.errorType)
+                    .setSubtype(Subtype.invalidPlaceHolder)
+                    .setMessage("Must have given, given2, surname, or surname2"));
+                return this; // fatal error, don't bother with others
+            }
+
+            // the rest of the tests are of the pattern, and only apply when we have both given and surname
+
+            if (firstGiven < Integer.MAX_VALUE && firstSurname < Integer.MAX_VALUE) {
+
+                Set<Order> order = parameterMatcher.getOrder();
+
+                // Handle 'sorting' value. Will usually be compatible with surnameFirst, except for known exceptions
+                // TODO Mark cleanup once sorting is in Order
```

Comment:
sorting is in Order now per #1869 (merged)

Reply:
done